### PR TITLE
change key to tag; add actual keyboard shortcuts; add menu item validate

### PR DESCRIPTION
**Pull request by crosebrugh**
Interested in your thoughts on my changes/additions. I didn't update the README or specs, but will do so if we can converge on something like these:
- change :key to :tag in order to use :key for the keyboard shortcut
- add keyboard shortcuts as a string
- add :validate in order to support validateMenuItem
- :target defaults to NSApp
- change MenuMotion::Menu update method to reconfig in order to not collide with NSMenu's update method

E.g.

```
{
  title: SHAPE_AXES_MENU_TITLE,
  target: self,
  action: :_static_axes,
  tag: :shape_axes,
  key: 'shift-cmd-y',
  validate: ->(mi) {
    mi.title = (@shape_view.axesStatic ?
                SHAPE_AXES_MENU_TITLE :
                STATIC_AXES_MENU_TITLE)
    @shape_view.axesVisible
  }
}                                               
```

### CODE REVIEW

**Comment by brianpattison** [comment]
@crosebrugh Hey, thanks for the patch! Here's some more info about how I'd like to approach some of this stuff:

### Keyboard Shortcuts

I like the idea of using `:tag` instead of `:key`, but I was actually thinking of using `:shortcut` and/or `:keyboard_shortcut` for the keyboard shortcuts to try to keep with the Formotion naming. Also, since it's `keyEquivalent` and `keyEquivalentModifierMask` on NSMenuItem, it's still confusing if `:key` matches up as a mashup of the two. So I guess I'm just thinking they should match up perfectly or just not match up at all. :)

``` ruby
# I'd personally prefer see this:
{
  title "Hello",
  key: :hello,
  shortcut: "cmd+h" # or "cmd-h"
}

# over this:
{
  title "Hello",
  tag: :hello,
  key: "cmd-h" # or "cmd+h"
}

# or this:
{
  title "Hello",
  tag: :hello,
  key: "h",
  key_modifier: "cmd"
}
```

**TL;DR:** I'd love for these to work: `shortcut: "cmd+h"` and `keyboard_shortcut: "cmd+h"`!

### NSMenuItem#target

Does `NSMenuItem#target` normally default to `NSApp`? If yes, that seems good.

### NSMenu#update

Good call... My bad on forgetting about the `#update` method. I would just make the naming more specific:

``` ruby
menu.update_item_with_key(:key, {
  title: "Hello"
})
```

### NSMenuItem Validation

I like the proposed syntax. Only thing I'd change for using as an example in the Readme is to use `menu_item` to make it more clear about what's getting passed into the Proc.

``` ruby
{
  title: "Hello World",
  validate: ->(menu_item) {
    true # or false
  }
}
```

I hope this helps move things in the right direction. I really need to get `MenuMotion::Menu#update` out of there ASAP, huh? :)

**Comment by crosebrugh** [comment]
That all looks good to me except for the :tag vs. :key. ObjectiveC uses
tags to grab elements so it seems like a good match to use that. Plus, :key
is confusing relative to having shortcut keys.

I like :tag and shortcut: 'cmd-h' (or 'cmd+h')

So, your final answer?

**Comment by brianpattison** [comment]
I can get behind `:tag` and `:shortcut`. There's probably no real reason to deprecate the use of `:key` since v0.1.0 will have to happen very soon anyways so we can also get rid of `MotionMenu::Menu#update`.

Of course the proposed `#update_item_with_key` would be updated as well:

``` ruby
menu.update_item_with_tag(:tag, {
  title: "Hello"
})
```

Thanks!

**Comment by crosebrugh** [comment]
pull request is updated as per our discussion.

fun!
